templates/lb-fargate-service/cf.yml

```diff
@@ -1,168 +1,171 @@
+  ContainerPort:
+    Type: Number
+    Default: {{.Image.Port}}
+  RulePath:
+    Type: String
+    Default: {{.Path}}
+  RulePriority:
+    Type: Number
+    Default: {{.PathPriority}} # This number must be unique between rules.
+  TaskCPU:
+    Type: String
+    Default: {{.CPU}}
+  TaskMemory:
+    Type: String
+    Default: {{.Memory}}
   TaskCount:
     Type: Number
-    Default: 2
-
+    Default: {{.Count}}
 Resources:
-  SecurityGroupEgress:
-    Type: AWS::EC2::SecurityGroupEgress
-    Properties:
-      GroupId:
-        Fn::ImportValue: !Sub ${EnvironmentName}-PublicLoadBalancerSecurityGroupId
-      IpProtocol: tcp
-      Description: Load balancer to target
-      DestinationSecurityGroupId: !GetAtt SecurityGroup.GroupId
-      FromPort: 80
-      ToPort: 80
-
-  Listener:
-    Type: AWS::ElasticLoadBalancingV2::Listener
-    Properties:
-      DefaultActions:
-        - TargetGroupArn: !Ref TargetGroup
-          Type: forward
-      LoadBalancerArn:
-        Fn::ImportValue: !Sub ${EnvironmentName}-PublicLoadBalancerArn
-      Port: 80
-      Protocol: HTTP
-
-  TargetGroup:
-    Type: AWS::ElasticLoadBalancingV2::TargetGroup
-    Properties:
-      Port: 80
-      Protocol: HTTP
-      TargetType: ip
-      VpcId:
-        Fn::ImportValue: !Sub ${EnvironmentName}-VpcId
-
-  TaskRole:
-    Type: AWS::IAM::Role
+  {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}LogGroup:
+    Type: AWS::Logs::LogGroup
     Properties:
-      AssumeRolePolicyDocument:
-        Statement:
-          - Action: sts:AssumeRole
-            Effect: Allow
-            Principal:
-              Service: ecs-tasks.amazonaws.com
-        Version: 2012-10-17
-
-  TaskDefinition:
+      LogGroupName: !Join ['', [/ecs/, !Ref ProjectName, '-', !Ref EnvName, '-', !Ref AppName]]
+  {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}TaskDefinition:
     Type: AWS::ECS::TaskDefinition
+    DependsOn: {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}LogGroup
     Properties:
+      Family: !Join ['', [!Ref ProjectName, '-', !Ref EnvName, '-', !Ref AppName]]
+      NetworkMode: awsvpc
+      RequiresCompatibilities:
+        - FARGATE
+      Cpu: !Ref TaskCPU
+      Memory: !Ref TaskMemory
+      ExecutionRoleArn: !Ref {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}ExecutionRole
+      TaskRoleArn: !Ref {{clean .ProjectName}}{{clean .EnvName}}{{clean .AppName}}TaskRole
       ContainerDefinitions:
-      # TODO: support sidecar images
-        - Essential: true
-          # TODO: support images from different repositories
-          Image: !Sub ${Image}
+        - Name: !Ref AppName
+          Image: !Ref ContainerImage
+          PortMappings:
+            - ContainerPort: !Ref ContainerPort {{if .Variables}}
+          Environment:{{range $name, $value := .Variables}}
+          - Name: {{$name}}
+            Value: {{$value}}{{end}}{{end}}{{if .Secrets}}
+          Secret:{{range $name, $valueFrom := .Screts}}
```

Comment:
`.Screts` -> `.Secrets`

Reply:
aah thanks! Tested Environment and Secrets too now